src/Database/LSMTree/Internal/Run.hs

```diff
@@ -141,3 +152,65 @@ fromWriteBuffer fs fsPaths buffer = do
     for_ (WB.content buffer) $ \(k, e) ->
       addFullKOp fs mrun k e
     fromMutable fs mrun
+
+-- | Load a previously written run from disk, checking each file's checksum
+-- against the checksum file.
+--
+-- TODO: define and throw exceptions instead of using 'error'?
```

Comment:
Yes, we'd want proper exceptions instead, which should also exported from the public API so the user can handle the exceptions if they want to

Reply:
e.g., something like [`ChecksumsFileFormatError`](https://github.com/input-output-hk/lsm-tree/blob/c6993d0f18757f08c8dc83c67cec2561fd24c049/src/Database/LSMTree/Internal/CRC32C.hs#L163-L166). I'd suggest adding the exceptions in this PR

Reply:
Done. There are lots of ways of structuring these, but I created two types for now for CRC error vs file format error.

Reply:
Looks good. My last suggestion would be to mention in the function's comment that one could expect exceptions

Reply:
Also: https://github.com/input-output-hk/lsm-tree/pull/111/files#r1522182511